daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/processors/unparsers/SeparatedSequenceUnparsers.scala

```diff
@@ -126,8 +126,26 @@ class ScalarOrderedSeparatedSequenceChildUnparser(
   override def unparse(state: UState) = childUnparser.unparse1(state)
 }
 
-class RepOrderedSeparatedSequenceChildUnparser(
-  childUnparser: Unparser,
+class ScalarOrderedSeparatedSequenceChildUnparser(childUnparser: Unparser,
+  srd: SequenceRuntimeData,
+  trd: TermRuntimeData,
+  sep: Unparser,
+  spos: SeparatorPosition,
+  ssp: SeparatorSuppressionPolicy,
+  ssAlgorithm: SeparatorSuppressionMode)
+  extends ScalarOrderedSeparatedSequenceChildUnparserBase(childUnparser, srd, trd, sep, spos, ssp, ssAlgorithm)
+
+class PotentiallyTrailingGroupSeparatedSequenceChildUnparser(childUnparser: Unparser,
```

Comment:
This class could be eliminated since it is identical to the preceeding. Where it is constructed could just do the 

new OtherClass with MixinTrait

trick. Not sure how much better I like that, but I hate these classes that just serve to forward all the constructor params.

Reply:
Seems like the mixin doesn't even really do much? Do you expect more stuff in here as this matures? Maybe it should just beome a boolean parameter that is checked in the right places?